### PR TITLE
Workaround llvm bug

### LINKED ISSUE #1076: llvm 3.7 bottle invalid CMake modules?
With llvm36 I just used `cmake .. -DLLVM_DIR=/usr/local/opt/llvm37/lib/llvm-3.7/share/llvm/cmake` and it worked like a charm but now it fails with:
```
CMake Error at /usr/local/opt/llvm37/lib/llvm-3.7/share/llvm/cmake/LLVMConfig.cmake:54 (include):
  include could not find load file:
    /LLVMExports.cmake
```
I compared LLVMConfig.cmake from 3.6 and 3.7 and there is a differene on how `LLVM_CMAKE_DIR` is set: `set(LLVM_CMAKE_DIR "${_LLVM_CMAKE_DIR}")` in 3.7, but I can't see there `${_LLVM_CMAKE_DIR}` is set and there is this comment at the top:
```CMake
# Compute the CMake directory from the LLVMConfig.cmake file location.n get_filename_component(_LLVM_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)n # Compute the installation prefix from the LLVMConfig.cmake file location.n get_filename_component(LLVM_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)n get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)n get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)n get_filename_component(LLVM_INSTALL_PREFIX "${LLVM_INSTALL_PREFIX}" PATH)n set(_LLVM_LIBRARY_DIR "${LLVM_INSTALL_PREFIX}/lib")
```
Os if I understood correctly in 3.7 these vars should be computed as stated by the comment but they aren't.

Do I understand this correctly? Can this be fixed? I'd really like if that'd be fixed for llvm 3.7.1

### PR DESCRIPTION
**Pull request by danilaml**
hopefully fix #1076

### CODE REVIEW

File: llvm37.rb [hunk 1]
```diff
@@ -123,6 +123,7 @@ class Llvm37 < Formula
   option "without-shared", "Don't build LLVM as a shared library"
   option "without-assertions", "Speeds up LLVM, but provides less debug information"
 
+  depends_on "gnu-sed"
```
Comment (mikemcquaid) [inline llvm37.rb]: This should be `=> :build` as it's not needed at runtime.

File: llvm37.rb [hunk 2]
```diff
@@ -269,6 +272,7 @@ def caveats; <<-EOS.undent
   end
 
   test do
+    assert_no_match /PATH\)n/, (lib/"llvm-3.7/share/llvm/cmake/LLVMConfig.cmake").read # test for sed errors
```
Comment (mikemcquaid) [inline llvm37.rb]: Mind adding the ` # test for sed errors` on the line above and expanding the description a bit? Thanks!

**Comment by mikemcquaid** [comment]
Hooray that this is working. A few final tweaks and I'll :ship: this.

**Comment by danilaml** [comment]
Rebased with suggested fixes.

**Comment by danilaml** [comment]
Anything else @mikemcquaid ?

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-versions/commit/5ca52f38742757b95b9661a1a0f1b56a8a8b6ef9. Thank you for your contribution to Homebrew @danilaml; we appreciate it! :smiley_cat:

**Pull request closed without merging**